packages/react-data-grid/src/Row.js

```diff
@@ -142,7 +142,14 @@ export default class Row extends React.Component {
   getExpandableOptions = (columnKey) => {
     const subRowDetails = this.props.subRowDetails;
     if (subRowDetails) {
-      return { canExpand: subRowDetails && subRowDetails.field === columnKey && ((subRowDetails.children && subRowDetails.children.length > 0) || subRowDetails.group === true), field: subRowDetails.field, expanded: subRowDetails && subRowDetails.expanded, children: subRowDetails && subRowDetails.children, treeDepth: subRowDetails ? subRowDetails.treeDepth : 0, subRowDetails };
+      return {
+        canExpand: subRowDetails && subRowDetails.field === columnKey && ((subRowDetails.children && subRowDetails.children.length > 0) || subRowDetails.group === true),
+        field: subRowDetails.field,
+        expanded: subRowDetails && subRowDetails.expanded,
+        hildren: subRowDetails && subRowDetails.children,
+        treeDepth: subRowDetails ? subRowDetails.treeDepth : 0,
+        subRowDetails
```

Comment:
This is wrapped in `if (subRowDetails) {`, so all these `subRowDetails &&` are redundant, as well as the `subRowDetails ?`

```js
getExpandableOptions = (columnKey) => {
  const { subRowDetails } = this.props;
  if (!subRowDetails) return {};

  const { field, expanded, children, treeDepth } = subRowDetails;

  return {
    canExpand: subRowDetails.field === columnKey && ((subRowDetails.children && subRowDetails.children.length > 0) || subRowDetails.group === true),
    field,
    expanded,
    children,
    treeDepth,
    subRowDetails
  };
};
```

You've also typo'd `children` as `hildren` 🤦‍♂️

Reply:
🤦‍♂️ 
- Lack of unit tests
- Need to move this to TS ASAP

Reply:
Fixed